Core/include/Acts/Navigation/DetectorVolumeFinders.hpp

```diff
@@ -36,11 +36,13 @@ struct RootVolumeFinder : public INavigationDelegate {
           "DetectorVolumeFinders: no detector set to navigation state.");
     }
 
-    const auto& volume = nState.currentDetector->rootVolume();
-    if (volume->inside(gctx, nState.position)) {
-      nState.currentVolume = volume;
-      volume->detectorVolumeUpdator()(gctx, nState);
-      return;
+    const auto& volumes = nState.currentDetector->rootVolumes();
+    for (const auto v : volumes) {
```

Comment:
Shouldn't that be `const auto&` (it's only a pointer in this case, but I think `clang-tidy` might be worried otherwise.

Reply:
I think this is okay because `auto` will resolve to a pointer and `&` would only bring another indirection